xbmc/addons/include/xbmc_pvr_types.h

```diff
@@ -255,28 +302,92 @@ extern "C" {
+    unsigned int iPreventDuplicateEpisodesDefault;          /*!< @brief (optional) The default value for PVR_TMER.iPreventDuplicateEpisodesSize. Must be filled if iPreventDuplicateEpisodesSize > 0 */
+
+    /* recording folder list value definitions */
+    unsigned int iRecordingGroupSize;                       /*< @brief (required) Count of possible values of PVR_TIMER.iRecordingGroup. 0 means folder lists are not supported by this timer type */
+    PVR_TIMER_TYPE_ATTRIBUTE_INT_VALUE
+      recordingGroup[PVR_ADDON_TIMERTYPE_VALUES_ARRAY_SIZE];
+                                                            /*!< @brief (optional) Array containing the possible values of PVR_TMER.iRecordingGroup. Must be filled if iRecordingGroupSize > 0 */
+    unsigned int iRecordingGroupDefault;                    /*! @brief (optional) The default value for PVR_TIMER.iRecordingGroup. Must be filled in if PVR_TIMER.iRecordingGroupSize > 0 */
+
+  } ATTRIBUTE_PACKED PVR_TIMER_TYPE;
+
+  /*!
    * @brief Representation of a timer event.
    */
   typedef struct PVR_TIMER {
     unsigned int    iClientIndex;                              /*!< @brief (required) the index of this timer given by the client */
-    int             iClientChannelUid;                         /*!< @brief (required) unique identifier of the channel to record on */
-    time_t          startTime;                                 /*!< @brief (required) start time of the recording in UTC. instant timers that are sent to the add-on by xbmc will have this value set to 0 */
-    time_t          endTime;                                   /*!< @brief (required) end time of the recording in UTC */
+    unsigned int    iParentClientIndex;                        /*!< @brief (optional) for timers scheduled by a repeating timer, the index of the repeating that scheduled this timer. 0 indicates "no parent". */
+    int             iClientChannelUid;                         /*!< @brief (optional) unique identifier of the channel to record on. For repeating timers, -1 will indicate "any" channel. */
+    time_t          startTime;                                 /*!< @brief (optional) start time of the recording in UTC. Instant timers that are sent to the add-on by Kodi will have this value set to 0. For repeatimg timers, 0 will indicate "any" time. */
+    time_t          endTime;                                   /*!< @brief (optional) end time of the recording in UTC. For repeating timers, 0 will indicate "any" time. */
     PVR_TIMER_STATE state;                                     /*!< @brief (required) the state of this timer */
-    char            strTitle[PVR_ADDON_NAME_STRING_LENGTH];    /*!< @brief (optional) title of this timer */
-    char            strDirectory[PVR_ADDON_URL_STRING_LENGTH]; /*!< @brief (optional) the directory where the recording will be stored in */
+    unsigned int    iTimerType;                                /*!< @brief (required) the type of this timer */
```

Comment:
ref to the types above and mention that this is a bitmask in the comment please. this api is exported to add-on authors, who you don't want guessing what these values should be ;-)